adagrams/game.py

```diff
@@ -1,11 +1,432 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+LETTER_SCORE = {
+    'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10
+}
+
+hand_size = 10
+
 def draw_letters():
-    pass
+    letter_list = []
+    hand = []
+
+    for key, value in LETTER_POOL.items():
+        letter_list.extend([key] * value)
+
+    for i in range(hand_size):
+        random_num = random.randint(0, len(letter_list) - 1)
+        hand.append(letter_list[random_num])
+        letter_list.pop(random_num)
+
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+    result = True
+    for letter in word:
+        if word.count(letter) > letter_bank.count(letter):
+            result = False
+            break
+    return result
 
 def score_word(word):
-    pass
+    total_value = 0
+    for letter in word.upper():
+        for key,value in LETTER_SCORE.items():
+            if letter == key:
+                total_value += value
```

Comment:
This will definitely work, but it will require you to loop through the keys every time. A more efficient approach might be to simply check to see if the letter is a valid key and if it is, access the value directly using LETTER_SCORE[key].